drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -405,15 +417,24 @@ local function find_default_endpoint(device)
   return device.MATTER_DEFAULT_ENDPOINT
 end
 
-local function assign_child_profile(device, child_ep)
+local function assign_child_profile(device, child_ep, ep_sequence)
   local profile
 
   -- check if device has an overridden child profile that differs from the profile
   -- that would match the child's device type
   for _, fingerprint in ipairs(child_device_profile_overrides) do
     if device.manufacturer_info.vendor_id == fingerprint.vendor_id and
        device.manufacturer_info.product_id == fingerprint.product_id then
-      return fingerprint.child_profile
+      if device.manufacturer_info.vendor_id == AQARA_MANUFACTURER_ID then
+        if ep_sequence == 1 then
+          -- To add Electrical Sensor only to the first EDGE_CHILD(light-power-energy-powerConsumption)
+          -- The profile of the second EDGE_CHILD is determined in the "for" loop below (e.g., light-binary)
+          device:set_field(IS_AQARA_SWITCH_DEVICE, true, {persist = true})
+          return fingerprint.child_profile
+        end
+      else
+        return fingerprint.child_profile
+      end
```

Comment:
```suggestion
      if device.manufacturer_info.vendor_id == AQARA_MANUFACTURER_ID then
        if ep_sequence == 1 then
          -- To add Electrical Sensor only to the first EDGE_CHILD(light-power-energy-powerConsumption)
          -- The profile of the second EDGE_CHILD is determined in the "for" loop below (e.g., light-binary)
          device:set_field(IS_AQARA_SWITCH_DEVICE, true, {persist = true})
        end
        return fingerprint.child_profile
      end
```
sorry, what I've been trying to say is that this edited implementation has the same handling, but is simpler. There's no reason to branch off and do an else statement.

Reply:
First, I would like to modify it as below. Because we have to consider the case below as well.
- if `device.manufacturer_info.vendor_id` is not `AQARA_MANUFACTURER_ID`, `fingerprint.child_profile` should be returned.
- If `ep_sequence == 2`, `fingerprint.child_profile` should not be returned. That is, if ‘ep_sequence == 2’, the ‘light-binary’ profile must be selected according to the for loop routine below.

```lua
    if device.manufacturer_info.vendor_id == fingerprint.vendor_id and
       device.manufacturer_info.product_id == fingerprint.product_id then
      if device.manufacturer_info.vendor_id == AQARA_MANUFACTURER_ID then
        if child_ep ~= 1 then
          -- To add Electrical Sensor only to the first EDGE_CHILD(light-power-energy-powerConsumption)
          -- The profile of the second EDGE_CHILD is determined in the "for" loop below (e.g., light-binary)
          break
        end
      end
      return fingerprint.child_profile
    end